Airwallex/Card/Internal/AWXCardViewModel.m

```diff
@@ -24,31 +29,73 @@ - (instancetype)initWithSession:(AWXSession *)session {
     return self;
 }
 
-- (void)saveBillingWithPlaceDetails:(AWXPlaceDetails *)placeDetails Address:(AWXAddress *)address completionHandler:(void (^)(AWXPlaceDetails *_Nullable, NSString *_Nullable))completionHandler {
-    placeDetails.address = address;
-    NSString *error = [placeDetails validate];
-    if (error) {
-        completionHandler(NULL, error);
-    } else {
-        completionHandler(address, NULL);
+- (BOOL)isBillingInformationRequired {
+    return self.session.requiresBilling;
+}
+
+#pragma mark Data validation
+
+- (void)validateSessionBillingWithError:(NSError **)error {
+    [self validateBillingDetailsWithPlace:self.session.billing andAddress:self.session.billing.address error:error];
+}
+
+- (void)validateBillingDetailsWithPlace:(AWXPlaceDetails *)placeDetails
+                             andAddress:(AWXAddress *)addressDetails
+                                  error:(NSError **)error {
+    self.validatedBilling = nil;
+    
+    if (!self.isBillingInformationRequired) {
+        return;
+    }
+    
+    AWXPlaceDetails *place = placeDetails.copy;
+    AWXAddress *address = addressDetails.copy;
+    place.address = address;
+    
+    *error = [place validate];
+    if (error == nil) {
+        self.validatedBilling = place;
```

Comment:
Nice work on creating new copy to submit request without potentially affecting the original reference passed from VC!

Reply:
I know I said "not right now" but couldn't help myself haha. Made things more organised in the view model too.